cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -313,14 +313,16 @@ impl Amode {
     ) {
         match self {
             Amode::ImmReg { base, .. } => {
-                collector.reg_use(*base);
+                if *base != regs::rbp() && *base != regs::rsp() {
+                    collector.reg_use(*base);
+                }
             }
             Amode::ImmRegRegShift { base, index, .. } => {
```

Comment:
We should assert here that base and index are not rbp or rsp I think, now that we're admitting we may use these registers in an `Amode`, just for consistency...